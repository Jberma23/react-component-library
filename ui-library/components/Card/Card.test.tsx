import { render, screen } from "@testing-library/react";
import { Card } from ".";

describe(Card, () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<Card>test card</Card>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", () => {
    render(<Card className="my-test-class">test card</Card>);
    const cardClass = screen.getByText("test card")?.getAttribute("class");
    expect(cardClass).toContain("my-test-class");
  });

  it("sets the tag name via the as prop", () => {
    render(<Card as="aside">test card</Card>);
    const card = screen.getByText("test card");
    expect(card).toHaveProperty("tagName", "ASIDE");
  });

  it("passes down additional props", async () => {
    render(
      <Card as="button" data-qa-id="hooray">
        test card
      </Card>
    );
    const cardAttribute = screen
      .getByRole("button")
      ?.getAttribute("data-qa-id");
    expect(cardAttribute).toEqual("hooray");
  });
});
