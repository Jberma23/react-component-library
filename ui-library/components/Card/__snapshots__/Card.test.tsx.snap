// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders with defaults 1`] = `
<DocumentFragment>
  <div
    class="rounded-3 transition duration-default bg-white shadow-0"
  >
    test card
  </div>
</DocumentFragment>
`;
